From: YueHaibing <yuehaibing@huawei.com>
Date: Thu, 11 Jul 2019 22:10:37 +0800
Subject: scsi: lpfc: Remove unnecessary null check before kfree
Patch-mainline: v5.4-rc1
Git-commit: 70a51d8c53176080e82b2aef050f820ccdd4b3f8
References: jsc#SLE-8284 jsc#SLE-8654

A null check before a kfree is redundant, so remove it.  This is detected
by coccinelle.

Signed-off-by: YueHaibing <yuehaibing@huawei.com>
Reviewed-by: James Smart <james.smart@broadcom.com>
Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
Acked-by: Daniel Wagner <dwagner@suse.de>
---
 drivers/scsi/lpfc/lpfc_bsg.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/drivers/scsi/lpfc/lpfc_bsg.c
+++ b/drivers/scsi/lpfc/lpfc_bsg.c
@@ -1276,9 +1276,7 @@ lpfc_bsg_hba_set_event(struct bsg_job *j
 	return 0; /* call job done later */
 
 job_error:
-	if (dd_data != NULL)
-		kfree(dd_data);
-
+	kfree(dd_data);
 	job->dd_data = NULL;
 	return rc;
 }
